go/internal/feast/registry/registry.go

```diff
@@ -93,312 +94,231 @@ func (r *Registry) RefreshRegistryOnInterval() {
 }
 
-func (r *Registry) loadOnDemandFeatureViews(registry *core.Registry) {
-	onDemandFeatureViews := registry.OnDemandFeatureViews
-	for _, onDemandFeatureView := range onDemandFeatureViews {
-		if _, ok := r.cachedOnDemandFeatureViews[r.project]; !ok {
-			r.cachedOnDemandFeatureViews[r.project] = make(map[string]*core.OnDemandFeatureView)
+func loadModels[U any, T any](protoList []U, cachedModels map[string]map[string]*model.ModelTTL[T], modelFactory func(proto U) T, project string, ttl time.Duration) {
+	for _, proto := range protoList {
+		if _, ok := cachedModels[project]; !ok {
+			cachedModels[project] = make(map[string]*model.ModelTTL[T])
+		}
+		obj := modelFactory(proto)
+		nameField := reflect.ValueOf(obj).Elem().FieldByName("Name")
+		if !nameField.IsValid() {
+			nameField = reflect.ValueOf(obj).Elem().FieldByName("Base").Elem().FieldByName("Name")
 		}
-		r.cachedOnDemandFeatureViews[r.project][onDemandFeatureView.Spec.Name] = onDemandFeatureView
+		objName := nameField.Interface().(string)
+		cachedModels[project][objName] = model.NewModelTTLWithExpiration(obj, ttl)
 	}
-}
 
-/*
-	Look up Entities inside project
-	Returns empty list if project not found
-*/
+}
 
-func (r *Registry) ListEntities(project string) ([]*model.Entity, error) {
+func (r *Registry) GetEntity(project string, entityName string) (*model.Entity, error) {
 	r.mu.RLock()
-	defer r.mu.RUnlock()
-	if cachedEntities, ok := r.cachedEntities[project]; !ok {
-		return []*model.Entity{}, nil
-	} else {
-		entities := make([]*model.Entity, len(cachedEntities))
-		index := 0
-		for _, entityProto := range cachedEntities {
-			entities[index] = model.NewEntityFromProto(entityProto)
-			index += 1
-		}
-		return entities, nil
+	cachedEntities, ok := r.cachedEntities[project]
+	r.mu.RUnlock()
+	if !ok && r.registryStore.HasFallback() {
+		return r.GetEntityFromRegistry(entityName, project)
+	} else if !ok {
+		return nil, fmt.Errorf("no cached entities found for project %s", project)
 	}
-}
 
-/*
-	Look up Feature Views inside project
-	Returns empty list if project not found
-*/
-
-func (r *Registry) ListFeatureViews(project string) ([]*model.FeatureView, error) {
-	r.mu.RLock()
-	defer r.mu.RUnlock()
-	if cachedFeatureViews, ok := r.cachedFeatureViews[project]; !ok {
-		return []*model.FeatureView{}, nil
-	} else {
-		featureViews := make([]*model.FeatureView, len(cachedFeatureViews))
-		index := 0
-		for _, featureViewProto := range cachedFeatureViews {
-			featureViews[index] = model.NewFeatureViewFromProto(featureViewProto)
-			index += 1
-		}
-		return featureViews, nil
+	cachedEntity, modelOk := cachedEntities[entityName]
+	if !modelOk && r.registryStore.HasFallback() {
+		return r.GetEntityFromRegistry(entityName, project)
+	} else if !ok {
+		return nil, fmt.Errorf("no cached entity %s found for project %s", entityName, project)
```

Comment:
Simplification - to avoid confusion. 
```suggestion
	if !modelOk {
	     if (r.registryStore.HasFallback()) {
		return r.GetEntityFromRegistry(entityName, project)
		} else {
		return nil, fmt.Errorf("no cached entity %s found for project %s", entityName, project)
		}
	}
```